### PR TITLE
Nrfx 4773 analyze persistence in npmx

### PR DESCRIPTION
**Pull request by wilu-nordic**

### CODE REVIEW

**Review by magp-nordic — APPROVED** [review]

File: drivers/npmx/Kconfig [hunk 1]
```diff
@@ -9,6 +9,11 @@ config NPMX
 
 if NPMX
 
+config NPMX_RESTORE_VALUES
+	bool "Restore values"
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply drivers/npmx/Kconfig:13]: The default is false.

[the remark this answers is not in the document]
Comment (wilu-nordic) [reply drivers/npmx/Kconfig:13]: The default value is `n` if not specified.

File: drivers/npmx/npmx_driver.c [hunk 2]
```diff
@@ -234,7 +234,12 @@ static int npmx_driver_init(const struct device *dev)
 
 	data->npmx_instance.generic_cb = generic_callback;
 
-	npmx_error_t ret = npmx_core_init(npmx_instance, backend, generic_callback);
+	bool restore_values = false;
+#ifdef CONFIG_NPMX_RESTORE_VALUES
+	restore_values = CONFIG_NPMX_RESTORE_VALUES;
+#endif
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply drivers/npmx/npmx_driver.c:240]: It con not be done like you want to. It the `NPMX_RESTORE_VALUES` is set to `n` or is not defined, in the `.config` file there is `# CONFIG_NPMX_RESTORE_VALUES is not set` so the symbol can not be used.

[the remark this answers is not in the document]
Comment (masz-nordic) [reply drivers/npmx/npmx_driver.c:240]: Then
```
#if defined(CONFIG_NPMX_RESTORE_VALUES)
npmx_error_t ret = npmx_core_init(npmx_instance, backend, generic_callback, true);
#else
npmx_error_t ret = npmx_core_init(npmx_instance, backend, generic_callback, false);
#endif
```

**Review by magp-nordic — APPROVED** [review]

**Review by magp-nordic — APPROVED** [review]

**Review by jaz1-nordic — DISMISSED** [review]

**Review by magp-nordic — APPROVED** [review]

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**